pkg/resources/resource_connection_aws.go

```diff
@@ -0,0 +1,275 @@
+	"comment":            CommentSchema(false),
+	"endpoint": {
+		Description: "Override the default AWS endpoint URL.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"region": {
+		Description: "The AWS region to connect to.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"access_key_id":     ValueSecretSchema("access_key_id", "The access key ID to connect with.", false),
+	"secret_access_key": IdentifierSchema("secret_access_key", "The secret access key corresponding to the specified access key ID.", false),
+	"session_token":     ValueSecretSchema("session_token", "The session token corresponding to the specified access key ID.", false),
+	"assume_role_arn": {
+		Description: "The Amazon Resource Name (ARN) of the IAM role to assume.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		ForceNew:    true,
+	},
+	"assume_role_session_name": {
+		Description:  "The session name to use when assuming the role.",
+		Type:         schema.TypeString,
+		Optional:     true,
+		ForceNew:     true,
+		RequiredWith: []string{"assume_role_arn"},
+	},
+	"validate":       ValidateConnectionSchema(),
+	"ownership_role": OwnershipRoleSchema(),
+}
+
+func ConnectionAws() *schema.Resource {
+	return &schema.Resource{
+		Description: "An Amazon Web Services (AWS) connection provides Materialize with access to an Identity and Access Management (IAM) user or role in your AWS account.",
+
+		CreateContext: connectionAwsCreate,
+		ReadContext:   connectionAwsRead,
+		UpdateContext: connectionAwsUpdate,
+		DeleteContext: connectionDelete,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: connectionAwsSchema,
+	}
+}
+
+func connectionAwsRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	i := d.Id()
+
+	metaDb, region, err := utils.GetDBClientFromMeta(meta, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	s, err := materialize.ScanConnectionAws(metaDb, utils.ExtractId(i))
+	if err == sql.ErrNoRows {
+		d.SetId("")
+		return nil
+	} else if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(utils.TransformIdWithRegion(string(region), i))
+
+	if err := d.Set("name", s.ConnectionName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("schema_name", s.SchemaName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("database_name", s.DatabaseName.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("endpoint", s.Endpoint.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("region", s.Region.String); err != nil {
+		return diag.FromErr(err)
+	}
+
+	// Note: Cannot set nested attributes with this SDK
+	// https://github.com/hashicorp/terraform-plugin-sdk/issues/459
+	// if err := d.Set("access_key_id.0.text", s.AccessKeyId.String); err != nil {
```

Comment:
Does this mean these attributes are not hooked up to anything? If so, we should probably comment them out so they don't show up in the docs?

Reply:
Nope, it's only in the read func, the create and update functions do use those attributes as normal. So this will mainly be a limitation if you try to import an existing AWS connection, the read func will not get those attributes during the import.

Reply:
This will be easy to fix once we migrate to the new framework as part of #225 though this work is yet to be planned.